File: solutions/1365/c++/solution.cpp
// 1365. How Many Numbers Are Smaller Than the Current Number
#include <vector>
#include <unsorted_map>
#include <algorithm>

using namespace std;

class Solution
{
	public:
	vector<int> smallerNumbersThanCurrent(vector<int>& nums)
	{
		vector<int> sortedNums(nums);

		sort(sortedNums.begin(), sortedNums.end());

		unordered_map<int, int> valuePosition;

		for (int i = nums.size() - 1; i >= 0; --i)
		{
			valuePosition[sortedNums[i]] = i;
		}

		vector<int> result(nums.size());

		for (int i = 0; i < nums.size(); ++i)
		{
			result[i] = valuePosition[nums[i]];
		}

		return result;
	}
};

File: solutions/1365/js/solution.js
// 1365. How Many Numbers Are Smaller Than the Current Number
var smallerNumbersThanCurrent = function(nums)
{
	let frequencies = new Array(101).fill(0);

	for (let i = 0; i < nums.length; ++i)
	{
		++frequencies[nums[i]];
	}

	let frequencySums = new Array(101);
	let sum = 0;

	for (let i = 0; i < frequencySums.length; ++i)
	{
		frequencySums[i] = sum;

		sum += frequencies[i];
	}

	let result = new Array(nums.length);

	for (let i = 0; i < nums.length; ++i)
	{
		result[i] = frequencySums[nums[i]];
	}

	return result;
};